sdk/tables/azure-data-tables/azure/data/tables/_base_client.py

```diff
@@ -102,32 +98,64 @@ def __init__(
                 account = parsed_url.netloc.split(".table.core.")
                 self.account_name = account[0] if len(account) > 1 else None
 
-        secondary_hostname = None
-        self.credential = credential
+        secondary_hostname: str = ""
+        self.credential: Optional[Union[AzureNamedKeyCredential, AzureSasCredential, TokenCredential]] = credential
         if self.scheme.lower() != "https" and hasattr(self.credential, "get_token"):
             raise ValueError("Token credential is only supported with HTTPS.")
-        if hasattr(self.credential, "named_key"):
-            self.account_name = self.credential.named_key.name  # type: ignore
+        if isinstance(self.credential, AzureNamedKeyCredential):
+            self.account_name = self.credential.named_key.name
             endpoint_suffix = os.getenv("TABLES_STORAGE_ENDPOINT_SUFFIX", DEFAULT_STORAGE_ENDPOINT_SUFFIX)
             secondary_hostname = f"{self.account_name}-secondary.table.{endpoint_suffix}"
 
-        if not self._hosts:
+        _hosts = kwargs.get("_hosts")
+        if not _hosts:
             if len(account) > 1:
                 secondary_hostname = parsed_url.netloc.replace(
                     account[0], account[0] + "-secondary"
                 ) + parsed_url.path.replace(account[0], account[0] + "-secondary").rstrip("/")
             if kwargs.get("secondary_hostname"):
                 secondary_hostname = kwargs["secondary_hostname"]
             primary_hostname = (parsed_url.netloc + parsed_url.path).rstrip("/")
-            self._hosts = {
+            _hosts = {
                 LocationMode.PRIMARY: primary_hostname,
                 LocationMode.SECONDARY: secondary_hostname,
             }
+        self._hosts = _hosts
 
-        self._policies = self._configure_policies(hosts=self._hosts, **kwargs)  # type: ignore
+        self._policies = self._configure_policies(hosts=self._hosts, **kwargs)
         if self._cosmos_endpoint:
             self._policies.insert(0, CosmosPatchTransformPolicy())
 
+        self._api_version = get_api_version(kwargs, "2019-02-02")
```

Comment:
I don't really like hard-coding the API version like this - if we ever regenerate to target a new default API version, no one will remember to update this here.

Reply:
Changed to use the default version from `self._client._config.version`